python/ql/src/semmle/python/dataflow/new/internal/DataFlowPublic.qll

```diff
@@ -135,9 +136,15 @@ class Node extends TNode {
   LocalSourceNode backtrack(TypeBackTracker t2, TypeBackTracker t) { t2 = t.step(result, this) }
 
   /**
-   * Gets a local source node from which data may flow to this node in zero or more local steps.
+   * Gets a local source node from which data may flow to this node in zero or more local data-flow steps.
    */
   LocalSourceNode getALocalSource() { result.flowsTo(this) }
+
+  /**
+   * Gets a local source node from which data may flow to this node in zero or more local taint-flow steps.
+   * WARNING: This will only have results if the taint tracking libraries have been imported into scope.
```

Comment:
This warning seems annoying. Is there anything blocking us from just adding a `private import TaintTrackingPublic`? -- which would make things work out of the box

Reply:
Indeed that imports all the concepts and actually changes test output (because all classes are also django helper classes).

Reply:
I have opened the relevant can of worms [on slack](https://github.slack.com/archives/CP6990YTU/p1615937236016100)...

Reply:
> actually changes test output (because all classes are also django helper classes).

What output is changed? (what tests are we talking about)

Reply:
We talked about this face-to-face. This approach had been taken previously, and the test changes can be seen in  changes from https://github.com/github/codeql/pull/5398/commits/b3ff3f7ee7f0838f43c29417033c89322b77dcb2 (where suddenly, classes are also treated as `DjangoViewClassHelper` when using `getAQlClass`). This stems from the fact that as part of `impoty python` we also privately import the dataflow library https://github.com/github/codeql/blob/7e3dbb0149bedc1a40e1531cf3a1be52bd4f0921/python/ql/src/python.qll#L40-L43, and with this change, we privately bring the taint tracking library into scope, which transitively brings all our framework modeling into scope.

@yoff and I talked a bit about the fact that importing taint-tracking would also import all external library models, so that all concrete subclasses of AdditionalTaintStep would also be in scope, following the advice in https://github.com/github/codeql/blob/main/docs/ql-design-patterns.md#importing-all-subclasses-of-a-class (that arguably, I wrote myself).

In the end, we opted for making `getALocalTaintSource` just works out of the box without having to remember to import something manually yourself.